### PR TITLE
Handle Break/Kill exceptions during #work_direct

### PR DESCRIPTION
**Pull request by ndbroadbent**
#130 plus a failing test

(needed this to make `any?` and `all?` much easier to implement)

### CODE REVIEW

File: lib/parallel.rb [hunk 1]
```diff
@@ -295,13 +295,18 @@ def add_progress_bar!(job_factory, options)
...
-          call_with_index(item, index, options, &block)
+      exception = nil
+      begin
+        while set = job_factory.next
+          item, index = set
+          results << with_instrumentation(item, index, options) do
+            call_with_index(item, index, options, &block)
+          end
         end
+      rescue StandardError => e
```
Comment (grosser) [inline lib/parallel.rb]: StandardError is the default ... so this should be the same:

```
rescue
  exception = $!
end
```

**Comment by grosser** [comment]
👍  super good ... these nasty little bugs :D

File: lib/parallel.rb (hunk 1 above)
Comment (ndbroadbent) [inline lib/parallel.rb]: Updated

**Comment by ndbroadbent** [comment]
I'll close this one since it's already in #199

**Pull request closed without merging**